### ISSUE TITLE
@EachNotBlank does not fail nulls

### ISSUE THREAD

**Issue by mbyrne00** [jirutka/validator-collection#15]
In `EachAnnotationIT.groovy`, at line 133 if you add another entry to your table it will fail

```EachNotBlank         | [:]                       | ['foo', 'bar']     | ['foo', null]```

Validator allows nulls through.

**Comment by jirutka** [comment]
That’s correct, the last column is _invalidValue_ and since `@NotBlank` permits nulls, `['foo', null]` is _valid_.

**Comment by jirutka** [comment]
Wait a moment… there’s something weird in the Hibernate Validator…

[NotBlank](https://github.com/hibernate/hibernate-validator/blob/6ae19900e2951513034534f9aec786689da9b4d8/engine/src/main/java/org/hibernate/validator/constraints/NotBlank.java#L25-L26):

> Validate that the annotated string is not `null` or empty.

[NotBlankValidator](https://github.com/hibernate/hibernate-validator/blob/6ae19900e2951513034534f9aec786689da9b4d8/engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/NotBlankValidator.java#L30-L31):

> Returns `true` if the string is `null` or the length of `charSequence` between the specified `min` and `max` values (inclusive), `false` otherwise.

**Comment by mbyrne00** [comment]
Hmm ... yea I went off the Javadoc ... I'll try a quick test of `@NotBlank` now

**Comment by mbyrne00** [comment]
`@NotBlank` definitely fails for `null` values. Looks like another instance where hibernate must chain two validators internally for an annotation.

**Comment by jirutka** [comment]
Ah, I see where's the problem. NotBlank is a composite constraint, it also includes NotNull (I've overlooked it before). So it's the same problem as before which I don't know how to *systematically* fix (see #8). I can solve it with workaround as in the previous issue though.

**Comment by mbyrne00** [comment]
Ah ok yea I saw that issue. Workaround would be great whenever you get time - thanks!

**Comment by jirutka** [comment]
OMG, I have already failing [test](https://github.com/jirutka/validator-collection/blob/579a6fb86d3b3c11ecc316ddf5abd1467b947ebb/src/test/groovy/cz/jirutka/validator/collection/CommonEachValidatorIT.groovy#L82-L118) for NotBlank! I totally forgot to it and don't have a clue why the heck I didn't fix it somehow.

**Comment by mbyrne00** [comment]
Ah yes, I saw those ignored tests and then saw you created other ones in `EachAnnotationIT` so I figured the other ones were specifically related to issue 8 and didn't understand at first glance what they were testing.

**Comment by mbyrne00** [comment]
On another note, that test with `@Ignore` entries seems to be testing failed indexes and the `EachAnnotationIT` one does not - this might be why https://github.com/jirutka/validator-collection/issues/16 occurred (no need to comment on that here ... separate issue for that).
